frontend/src/Metamaps/DataModel/Mapping.js

```diff
@@ -0,0 +1,37 @@
+import _ from 'lodash'
+import Backbone from 'backbone'
+try { Backbone.$ = window.$ } catch (err) {}
+
+import Active from '../Active'
+import Map from '../Map'
+import Synapse from '../Synapse'
+import Topic from '../Topic'
+
+const Mapping = Backbone.Model.extend({
+  urlRoot: '/mappings',
+  blacklist: ['created_at', 'updated_at'],
+  toJSON: function (options) {
+    return _.omit(this.attributes, this.blacklist)
+  },
+  initialize: function () {
+    if (this.isNew()) {
+      this.set({
+        'user_id': Active.Mapper.id,
+        'map_id': Active.Map ? Active.Map.id : null
+      })
+    }
+  },
+  getMap: function () {
+    return Map.get(this.get('map_id'))
```

Comment:
great!

One thing about es6 promises: they need a polyfill. I considered polyfills, but eventually realized jQuery has promises built in and we're already using them. See Import.js for how I used them (with $.Deferred()). It's a bit clunkier than es6 promises, but it's not really that bad for our simple use cases.